hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -37,6 +38,29 @@
  */
 public class HoodieIndexUtils {
 
+  /**
+   * Fetches Pair of partition path and {@link HoodieBaseFile}s for interested partitions.
+   *
+   * @param partition   Partition of interest
+   * @param context     Instance of {@link HoodieEngineContext} to use
+   * @param hoodieTable Instance of {@link HoodieTable} of interest
+   * @return the list of {@link HoodieBaseFile}
+   */
+  public static List<HoodieBaseFile> getLatestBaseFilesForPartition(
```

Comment:
shall we use this in `getLatestBaseFilesForAllPartitions` to avoid duplicate codes.

Reply:
I think we could.

Reply:
It's good that `getLatestBaseFilesForPartition` was extracted from `getLatestBaseFilesForAllPartitions`.  

Current codebase: 
```
  public static List<HoodieBaseFile> getLatestBaseFilesForPartition(
      final String partition,
      final HoodieTable hoodieTable) {
    Option<HoodieInstant> latestCommitTime = hoodieTable.getMetaClient().getCommitsTimeline()
        .filterCompletedInstants().lastInstant();
    if (latestCommitTime.isPresent()) {
      return hoodieTable.getBaseFileOnlyView()
          .getLatestBaseFilesBeforeOrOn(partition, latestCommitTime.get().getTimestamp())
          .collect(toList());
    }
    return Collections.emptyList();
  }
```

Maybe the following implementation is more efficient
```
  public static List<HoodieBaseFile> getLatestBaseFilesForPartition(
      final String partition,
      final HoodieTable hoodieTable) {
    return hoodieTable.getFileSystemView()
        .getAllFileGroups(partition)
        .map(HoodieFileGroup::getLatestDataFile)
        .filter(Option::isPresent)
        .map(Option::get)
        .collect(toList());
  }
```

Reply:
refer to `SparkHoodieBackedTableMetadataWriter#prepRecords`

Reply:
Agree, there is no need to decide and compare the instant time here, but i would not promote it in this PR, because it is not related.

You can promote it in a separate JIRA issue.